src/sentry/snuba/query_subscription_consumer.py

```diff
@@ -166,6 +169,9 @@ def handle_message(self, message):
                         "value": message.value(),
                     },
                 )
+                _delete_from_snuba(
+                    self.topic_to_dataset[message.topic()], contents["subscription_id"]
```

Comment:
Could there be other remnants in the database for the alert rule or rule triggers that also need to be cleaned up?

Reply:
If there are then there's no way for us to find them without a `QuerySubscription`. This shouldn't be the case in general due to how we handle deletes - this is mostly to handle the case where we retry POST and create multiple subscriptions, and also if there are any problems when updating an alert rule.